src/test/java/seedu/address/logic/commands/EditGradedTestCommandTest.java

```diff
@@ -0,0 +1,201 @@
+        expectedModel.setPerson(model.getFilteredPersonList().get(0), editedPerson);
+
+        // Check if the command execution is successful
+        assertCommandSuccess(editGradedTestCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void equals() {
+        Index targetIndex = INDEX_FIRST_PERSON;
+
+        EditGradedTestCommand editGradedTestCommandWithRA1 = new EditGradedTestCommand(
+                targetIndex,
+                Optional.of(new ReadingAssessment1("10.0")),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty());
+
+        EditGradedTestCommand editGradedTestCommandWithRA2 = new EditGradedTestCommand(
+                targetIndex,
+                Optional.empty(),
+                Optional.of(new ReadingAssessment2("20.0")),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty());
+
+        EditGradedTestCommand editGradedTestCommandWithMidTerms = new EditGradedTestCommand(
+                targetIndex,
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(new MidTerms("70.0")),
+                Optional.empty(),
+                Optional.empty());
+
+        EditGradedTestCommand editGradedTestCommandWithFinals = new EditGradedTestCommand(
+                targetIndex,
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(new Finals("100.0")),
+                Optional.empty());
+
+        EditGradedTestCommand editGradedTestCommandWithPE = new EditGradedTestCommand(
+                targetIndex,
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(new PracticalExam("100.0")));
+
+        EditGradedTestCommand editGradedTestCommandDifferentIndex = new EditGradedTestCommand(
+                INDEX_SECOND_PERSON,
+                Optional.of(new ReadingAssessment1("10.0")),
+                Optional.of(new ReadingAssessment2("20.0")),
+                Optional.of(new MidTerms("70.0")),
+                Optional.of(new Finals("100.0")),
+                Optional.of(new PracticalExam("100.0")));
+
+        EditGradedTestCommand editGradedTestCommandSameValues = new EditGradedTestCommand(
+                targetIndex,
+                Optional.of(new ReadingAssessment1("10.0")),
+                Optional.of(new ReadingAssessment2("20.0")),
+                Optional.of(new MidTerms("70.0")),
+                Optional.of(new Finals("100.0")),
+                Optional.of(new PracticalExam("100.0")));
+
+        // Test for equality
+        assertTrue(editGradedTestCommandWithRA1.equals(editGradedTestCommandWithRA1));
+        assertTrue(editGradedTestCommandWithRA1.equals(new EditGradedTestCommand(
```

Comment:
Consider using assertEquals

Reply:
rayson said if we use assertEqual, the codecov cant catch the coverage or smt haha